frontend/lib/algolia/server/findDescendantDevicesWithProducts.ts

```diff
@@ -0,0 +1,39 @@
+import {
```

Comment:
I suggest keeping this file within the product list model folder, since this is product list related business logic

Reply:
https://github.com/iFixit/react-commerce/pull/1252/commits/bdcfbba626eb7f4a0840febdd71c0f0ad3131539